fdbserver/include/fdbserver/IKeyValueStore.h

```diff
@@ -167,7 +167,7 @@ extern IKeyValueStore* keyValueStoreLogSystem(class IDiskQueue* queue,
                                               bool disableSnapshot,
                                               bool replaceContent,
                                               bool exactRecovery,
-                                              bool enableEncryption);
+                                              EncryptionAtRestMode encryptMode);
```

Comment:
Add comment note the KeyValueStoreLogSystem will encrypt everything in system key domain regardless of key prefixes and the specific encryption mode (domain-aware/cluster-aware)?

I would prefer keeping the enableEncryption boolean flag since it is less confusing.

Reply:
Hmm i see im assuming the reason we do this is because everything is a system key in the IKeyValueStore? In that case should we assert this is the case? I would like to keep the encryptMode consistent throughout the codebase (instead of using a bool).

Reply:
right, the current logic assume KVStoreMemory encryption is used only by the txnStateStore, and not by, for example, the storage server. I'm fine if you could describe the behavior here as inline comment, but reverting to the boolean flag would be less confusing and won't add much effort.

Reply:
I feel the use of `EncryptionAtRestMode` being confusing here because the KVStoreMemory support neither domain-aware nor cluster-aware mode.

Reply:
Sure i can keep it as a bool then